test/controllers/changeset_comments_controller_test.rb

```diff
@@ -46,6 +46,14 @@ def test_feed
         assert_select "item", :count => 3
       end
     end
+
+    # Gets comment Ids from HTML and checks that they are in descending order
+    last_comment_id = -1
+    @response.body.split("Changeset comment ")[1..].each_with_index do |data, index|
+      id = Integer data.split("<")[0]
```

Comment:
The normal way to convert a string to an integer is to use `.to_i` rather than calling the constructor like this.

Reply:
Thanks for comments. I agree with you, but when I parsed XML, the only id info was still in the CDATA containing HTML (fields exposed in the XML were identical for every comment e.g. comment creation time, because they are created with no delay). So even if I used parsed data, I would still use HTML raw string manipulation. Certainly, it's doable (and at first I did it that way), but it makes the test more complicated (so I rewrote it this way). If it's okay, test to be more complicated, I'll do it with XML parser, but I think this solution is already more complicated than the regular test should be.

Reply:
I rewrote the test in a different way